cvat-core/src/server-proxy.js

```diff
@@ -619,19 +624,91 @@
+                        } catch (errorData) {
+                            try {
+                                await deleteTask(response.data.id);
+                            } catch (_) {
+                                // ignore
+                            }
+                            throw generateError(errorData);
+                        }
                     }
+                    await upload();
+                }
 
-                    throw generateError(errorData);
+                async function bulkUpload(files) {
+                    const requestLimit = 1024 * 1024 * 100; // 100 mb
+                    const fileGroups = [];
+                    let currentGroup = [];
+                    let currentGroupSize = 0;
+                    for (const file of files) {
+                        if (requestLimit - currentGroupSize >= file.size) {
+                            currentGroup.push(file);
+                            currentGroupSize += file.size;
+                        } else {
+                            fileGroups.push(currentGroup);
+                            currentGroupSize = file.size;
+                            currentGroup = [file];
+                        }
+                    }
+                    fileGroups.push(currentGroup);
+                    const totalChunks = fileGroups.length;
+                    let currentChunkNumber = 1;
+                    async function upload() {
+                        if (currentChunkNumber > 1) {
+                            fileGroups[currentChunkNumber - 2].forEach((element, idx) => {
+                                taskData.delete(`client_files[${idx}]`);
+                            });
+                        }
+                        fileGroups[currentChunkNumber - 1].forEach((element, idx) => {
+                            taskData.set(`client_files[${idx}]`, element);
+                        });
+                        taskData.set('end_of_upload', currentChunkNumber >= totalChunks);
+                        taskData.set('use_chunk_upload', false);
+                        try {
+                            await Axios.post(`${backendAPI}/tasks/${response.data.id}/data`, taskData, {
+                                proxy: config.proxy,
+                            });
+                            onUpdate(`The data are being uploaded to the server ${Math.round((currentChunkNumber / totalChunks) * 100)}%`);
+                            currentChunkNumber++;
+                            if (currentChunkNumber <= totalChunks) await upload();
+                        } catch (errorData) {
+                            try {
+                                await deleteTask(response.data.id);
+                            } catch (_) {
+                                // ignore
+                            }
+                            throw generateError(errorData);
+                        }
```

Comment:
The same comment about try block. Better to put it outside.

Reply:
Done